File: Bubble/Bubble.js
/**
 * 버블정렬 수행
 * 
 * @param {Array<Number>} arr 정렬할 배열 
 */
function bubbleSort(arr) {
    for(let i = 0; i < arr.length-1; i++) {
        for(let j = 0; j < arr.length-i-1; j++) {
            if(arr[j] > arr[j+1]) {
                swap(arr, j, j+1);
            }
        }
    }
}

/**
 * arr의 idx1과 idx2의 인덱스에 해당하는 두 원소 서로 교체
 * 
 * @param {Array<Number>} arr 스왑할 배열
 * @param {Number} idx1 1번째 인덱스
 * @param {Number} idx2 2번째 인덱스
 */
function swap(arr, idx1, idx2) {
    let tmp = arr[idx2];
    arr[idx2] = arr[idx1];
    arr[idx1] = tmp;
}

let a = [1,5,9,3,2,4,7,8,6,0];
bubbleSort(a);

console.log(a);

File: Bubble/Bubble.java
import java.util.*;

public class Main {
    private static final int ARRAY_SIZE = 1000;

    public static void main(String[] args) {
        Integer[] arr = new Integer[ARRAY_SIZE];
        for(int i = 0; i < ARRAY_SIZE; i++) {
            arr[i] = i;
        }
        Collections.shuffle(Arrays.asList(arr));
        int[] newArr = Arrays.stream(arr).mapToInt(Integer::intValue).toArray();
        bubbleSort(newArr);

        for(int i : newArr) {
            System.out.println(i);
        }
    }

    public static void bubbleSort(int[] arr) {
        for(int i = 0; i < arr.length-1; i++) {
            for(int j = 0; j < arr.length-i-1; j++) {
                if(arr[j] > arr[j+1]) {
                    swap(arr, j, j+1);
                }
            }
        }
    }

    public static void swap(int[] arr, int idx1, int idx2) {
        int tmp = arr[idx2];
        arr[idx2] = arr[idx1];
        arr[idx1] = tmp;
    }
}